product_sequence/models/product_category.py

```diff
@@ -21,6 +21,15 @@ class ProductCategory(models.Model):
         copy=False, readonly=True,
     )
 
+    @api.model
+    def get_sequence_id(self):
```

Comment:
I think it is better to use the decorater @api.multi with self.ensure_one() or @api.one

Reply:
@api.one is deprecated and with ensure_one the following does not work:
```python
self.env['product.category'].get_sequence_id()
```

Or do you mean it like that?
```python
@api.multi
def get_sequence_id(self):
    default_sequence = self.env.ref('product_sequence.seq_product_auto')
    if self:
        self.ensure_one()
        current_category = self
        if self.env.user.company_id.use_parent_categories_to_determine_prefix:
            while not current_category.sequence_id and current_category.parent_id:
                current_category = current_category.parent_id
        return current_category.sequence_id or default_sequence
    else:
        return default_sequence
```

Reply:
I know what you want, But if you define the function with "@api.model". It confuse me that It can only be called from self.env['product.category']